### ISSUE TITLE
Undefined reference for NodeHandle::getParamCached() float overload

### ISSUE THREAD

**Issue by theidealist** [ros/ros_comm#623]
A simple test like:

```
ros::NodeHandle nh;
float test;
nh.getParamCached("cheese",test);
```

Fails at link time due to missing implementation for the publicly advertised float overload of NodeHandle::getParamCached().

**Comment by dirk-thomas** [comment]
Thanks for the report.

Depending on how the discussion on #621 will go I will:
- either add the missing definition of the function to implement the existing declaration
- or remove the not implemented declaration form the header file

**Comment by dirk-thomas** [comment]
Addressed by #635.
